### PR TITLE
Ldap change notifications

### PR DESCRIPTION
**Pull request by kthblmfld**
Squashed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,13 +208,15 @@ hanging in the legacy UI and stack traces from rbac_server:
...
 
-    docker-compose down
+    bin/stop
 
     docker rm -vf $(docker ps -a -q)
 
     docker rmi -f $(docker images -a -q)
 
-    docker-compose up --build
+    bin/build
```
Comment (adamgering) [inline README.md]: bin/build doesn't build docker containers, and requires flags.

bin/start -b starts the docker containers with the --build options.

File: docker-compose.yaml [hunk 2]
```diff
@@ -57,7 +57,25 @@ services:
       - ledger-sync
     command: ./bin/rbac-providers-azure
 
-  ldap-rbac-provider:
+  rbac-provider-azure:
```
Comment (adamgering) [inline docker-compose.yaml]: Looks like Azure has been added twice.

File: docker/rbac.providers.ldap.Dockerfile [hunk 3]
```diff
@@ -33,16 +33,13 @@ RUN pip3 install \
         grpcio-tools \
         itsdangerous \
         rethinkdb \
-        sanic==0.7.0
+        sanic==0.7.0 \
+        tornado==4.5.3 \
```
Comment (adamgering) [inline docker/rbac.providers.ldap.Dockerfile]: Non-shared dependencies go outside the common block where they were before this change.

File: docker/rbac.providers.ldap.Dockerfile [hunk 4]
```diff
@@ -33,16 +33,13 @@ RUN pip3 install \
...
 WORKDIR /project/hyperledger-rbac
-# -----------------------------------------------------------------------------
-# End base docker image config for Hyperledger RBAC Next Directory
-# -----------------------------------------------------------------------------
 
-RUN pip3 install \
-        rethinkdb \
-        ldap3 \
-        pyasn1==0.4.4
+COPY . .
```
Comment (adamgering) [inline docker/rbac.providers.ldap.Dockerfile]: Volume mounts, not copy.

File: rbac/providers/ldap/ldap_sync.py [hunk 5]
```diff
@@ -107,9 +107,9 @@ def insert_to_db(data_dict, data_type):
             "timestamp": datetime.now().replace(tzinfo=timezone.utc).isoformat(),
             "provider_id": LDAP_DC
         }
-        r.table("inbound_queue").insert(inbound_entry).run()
+        r.table("queue_inbound").insert(inbound_entry).run()
 
-    print("Inserted %s records into inbound_queue." % len(data_dict))
+    print("Inserted %s records into queue_inbound." % len(data_dict))
```
Comment (adamgering) [inline rbac/providers/ldap/ldap_sync.py]: LOGGER, not print.

File: bin/setup_db [hunk 6]
```diff
@@ -64,12 +67,18 @@ def setup_db(host, port, name):
...
 
-        print('Creating table: inbound_queue')
-        r.db(name).table_create('inbound_queue').run(conn)
+        print('Creating table: queue_inbound')
+        r.db(name).table_create('queue_inbound').run(conn)
 
         print('Creating table: sync_tracker')
         r.db(name).table_create('sync_tracker').run(conn)
 
+        print('Creating table: queue_inbound')
```
Comment (nadiabahrami) [inline bin/setup_db]: duplicate of lines above ln 70, 71

File: docker-compose.yaml (hunk 2 above)
Comment (kthblmfld) [reply docker-compose.yaml]: Removed

File: docker/rbac.providers.ldap.Dockerfile (hunk 4 above)
Comment (kthblmfld) [reply docker/rbac.providers.ldap.Dockerfile]: Done

File: docker/rbac.providers.ldap.Dockerfile (hunk 3 above)
Comment (kthblmfld) [reply docker/rbac.providers.ldap.Dockerfile]: Ah, Moved them back with a meaningful comment explaining why we do this

File: rbac/providers/ldap/ldap_sync.py (hunk 5 above)
Comment (kthblmfld) [reply rbac/providers/ldap/ldap_sync.py]: Meh why not, as long as I'm in this file. Replaced usages of print with logger

**Comment by kthblmfld** [comment]
The linter error was caused by having a TODO in the changes. I removed it hoping we find time to revisit the linting rules (active and disabled) in the near future.

**Comment by adamgering** [comment]
Can we rebase and squash so this doesn't show 11 commits?

**Comment by kthblmfld** [comment]
Commits squashed

**Comment by kthblmfld** [comment]
Looks like pylint and black are not applying the same standards. I merged in a change master of a file that was formatted by black, and pylint failed the continuation indent.

**Pull request merged**